synchronous-replication.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Synchronous Replication for Dual Data Centers
+summary: Learn how to configure synchronous replication.
+---
+
+# Synchronous Replication for Dual Data Centers
+
+This document introduces how to configure synchronous replication for dual data centers.
+
+> **Warning:**
+>
+> Synchronous replication is still an experimental feature. Do not use it in a production environment.
+
+In the scenario of dual data centers, one is the primary center and the other is the DR (data recovery) center. When a Region has an odd number of replicas, more replicas are placed in the primary center. When the DR center is down for more than a specified period of time, the asynchronous mode is used by default for the replication between two centers.
+
+To use the synchronous mode, you can configure in the PD configuration file or change the replication mode manually using pd-ctl.
+
+## Enable synchronous replication in PD configuration file
+
+The replication mode is controlled by PD. You can configure in the PD configuration file when deploying a cluster. See the following example:
+
+{{< copyable "" >}}
+
+```toml
+[replication-mode]
+replication-mode = "dr-auto-sync"
+[replication-mode.dr-auto-sync]
+label-key = "zone"
+primary = "z1"
+dr = "z2"
+primary-replicas = 2
+dr-replicas = 1
+wait-store-timeout = "1m"
+wait-sync-timeout = "1m"
+```
+
+In the configuration above:
+
++ `dr-auto-sync` is the mode to enable synchronous replication.
++ The label key `zone` is used to distinguish different data centers.
++ TiKV instances with the `"z1"` value are considered the primary data center, and TiKV instances with `"z2"` are the DR data center.
++ `primary-replicas` is the number of replicas that should be placed in the primary data center.
++ `dr-replicas` is the number of replicas that should be placed in the DR data center.
++ `wait-store-timeout` is the time to wait before falling back to asynchronous replication.
++ `wait-sync-timeout` is the time to wait before forcing TiKV to change replication mode (currently not supported).
+
+To check the current replication state of the cluster, use the following URL:
+
+{{< copyable "shell-regular" >}}
+
+```bash
+% curl http://pd_ip:pd_port/pd/api/v1/replication_mode/status
+```
+
+```bash
+{
+  "mode": "dr-auto-sync",
+  "dr-auto-sync": {
+    "label-key": "zone",
+    "state": "sync"
+  }
+}
+```
+
+> **Note:**
+>
+> The replication mode indicates how a single Region is replicated, either `asynchronous` or `synchronous`. The replication state of the cluster indicates how all Regions are replicated, with the options of `async`, `sync-recover`, and `sync`.
```

Comment:
@BusyJay I wrote it based on our discussion. PTAL whether I have any misunderstanding, thanks!

Reply:
The first sentence is wrong. `mode` can only be `dr-auto-sync` or `majority`. I think you can just delete the sentence.

Reply:
Deleted.
```suggestion
> The replication state of the cluster indicates how all Regions are replicated, with the options of `async`, `sync-recover`, and `sync`.
```